security/crypto/bls12_test.go

```diff
@@ -0,0 +1,219 @@
+package crypto_test
+
+import (
+	"testing"
+
+	"github.com/relab/hotstuff"
+	"github.com/relab/hotstuff/core"
+	"github.com/relab/hotstuff/internal/testutil"
+	"github.com/relab/hotstuff/security/crypto"
+)
+
+// createBLS12Signers creates BLS12-based signers for testing.
+// Returns a slice of signers, one for each replica.
+func createBLS12Signers(t testing.TB, numReplicas int) []crypto.Base {
+	t.Helper()
+	// Generate keys for all replicas
+	keys := make(map[hotstuff.ID]hotstuff.PrivateKey, numReplicas)
+	for i := 1; i <= numReplicas; i++ {
+		keys[hotstuff.ID(i)] = testutil.GenerateBLS12Key(t)
+	}
+	
+	// Create configs for all replicas
+	configs := make([]*core.RuntimeConfig, numReplicas)
+	for i := 1; i <= numReplicas; i++ {
```

Comment:
Since `i-1` is used to index into the slice, we should change this to:
```go
for i := range numReplicas {
```
And then update to `hotstuff.ID(i+1)` and similar for the `j` loop below.